lib/python/qmk/painter.py

```diff
@@ -79,6 +80,110 @@
     }
 }
 
+
+def _render_text(values):
+    # FIXME: May need more chars with GIFs containing lots of frames (or longer durations)
+    return "|".join([f"{i:4d}" for i in values])
+
+
+def _render_numeration(metadata):
+    return _render_text(range(len(metadata)))
+
+
+def _render_values(metadata, key):
+    return _render_text([i[key] for i in metadata])
+
+
+def _render_image_metadata(metadata):
+    size = metadata.pop(0)
+
+    lines = [
+        "// Image's metadata",
+        "// ----------------",
+        f"// Width: {size['width']}",
+        f"// Height: {size['height']}",
+    ]
+
+    if len(metadata) == 1:
+        lines.append("// Single frame")
+
+    else:
+        lines.extend([
+            f"//        Frame: {_render_numeration(metadata)}",
+            f"// Duration(ms): {_render_values(metadata, 'delay')}",
+            f"//  Compression: {_render_values(metadata, 'compression')} >> See qp.h, painter_compression_t",
+            f"//        Delta: {_render_values(metadata, 'delta')}",
+        ])
+
+        deltas = []
+        for i, v in enumerate(metadata):
+            # Not a delta frame, go to next one
+            if not v["delta"]:
+                continue
+
+            # Unpack rect's coords
+            l, t, r, b = v["delta_rect"]
+
+            delta_px = (r - l) * (b - t)
+            px = size["width"] * size["height"]
+
+            # FIXME: May need need more chars here too
+            deltas.append(f"// Frame {i:3d}: ({l:3d}, {t:3d}) - ({r:3d}, {b:3d}) >> {delta_px:4d}/{px:4d} pixels ({100*delta_px/px:.2f}%)")
+
+        if deltas:
+            lines.append("// Areas on delta frames")
+            lines.extend(deltas)
+
+    return "\n".join(lines)
+
+
+def generate_subs(cli, out_bytes, *, font_metadata=None, image_metadata=None):
+    if font_metadata is not None and image_metadata is not None:
+        raise ValueError("Cant generate subs for font and image at the same time")
+
+    # Work out the command that was executed
+    args = [arg for arg in cli.args.__dict__ if not arg.startswith("_") and arg not in ("datetime_fmt", "log_fmt", "log_file_fmt", "log_file_level", "log_file", "color", "interactive", "config_file", "subparsers", "entrypoint")]
+    args_str = " ".join((f"--{arg} {getattr(cli.args, arg)}" for arg in sorted(args)))
+    command = f"qmk {cli.args.subparsers} {args_str}"
```

Comment:
This wouldnt be accepted, the context of what command was ran should just be passed through to the function

Reply:
I made this hacky way just so this updates itself when a command/arg is added/renamed.

Would this kind of code be accepted at the command's function (eg: `painter_convert_graphics`) or should it be hardcoded in?